scripts/exregional_run_fcst.sh

```diff
@@ -449,18 +455,96 @@ cat > itag <<EOF
 /
 EOF
 fi
+#
+#-----------------------------------------------------------------------
+#
+# Update or copy the FV3 input.nml file
+#
+#-----------------------------------------------------------------------
+#
+if [ "${DO_ENSEMBLE}" = TRUE ] && ([ "${DO_SPP}" = TRUE ] || [ "${DO_SPPT}" = TRUE ] || [ "${DO_SHUM}" = TRUE ] || \
+   [ "${DO_SKEB}" = TRUE ] || [ "${DO_LSM_SPP}" =  TRUE ]); then
+  python3 $USHdir/set_FV3nml_ens_stoch_seeds.py \
+      --path-to-defns ${GLOBAL_VAR_DEFNS_FP} \
+      --cdate "$CDATE" || print_err_msg_exit "\
+Call to function to create the ensemble-based namelist for the current
+cycle's (cdate) run directory (DATA) failed:
+  cdate = \"${CDATE}\"
+  DATA = \"${DATA}\""
+else
+  cp_vrfy "${FV3_NML_FP}" "${DATA}/${FV3_NML_FN}"
+fi
+#
+#-----------------------------------------------------------------------
+#
+# Replace parameter values for restart in FV3 input.nml and model_configure.
+# Add restart files to INPUT directory.
+#
+#-----------------------------------------------------------------------
+#
+flag_fcst_restart="FALSE"
+if [ "${DO_FCST_RESTART}" = "TRUE" ] && [ "$(ls -A ${DATA}/RESTART )" ]; then
+  cp_vrfy input.nml input.nml_org
+  cp_vrfy model_configure model_configure_org
+  if [ "${CPL_AQM}" = "TRUE" ]; then
+    cp_vrfy aqm.rc aqm.rc_org
+  fi
+  relative_link_flag="FALSE"
+  flag_fcst_restart="TRUE"
 
-if [ "${CPL_AQM}" = "TRUE" ]; then
+  # Update FV3 input.nml for restart
+  python3 $USHdir/update_restart_input_nml_file.py \
+    --path-to-defns ${GLOBAL_VAR_DEFNS_FP} \
+    --run_dir "${DATA}" \
+    || print_err_msg_exit "\
+Call to function to update the FV3 input.nml file for restart for the 
+current cycle's (cdate) run directory (DATA) failed:
+  cdate = \"${CDATE}\"
+  DATA = \"${DATA}\""
+
+  # Search for available restart files and find the latest hour
+  fnm_rst_pdy=()
+  fnm_rst_hms=() 
+  for fnm_rst in "${DATA}/RESTART"/*; do
+    fnm_rst_base=$( basename "${fnm_rst}" )
+    fnm_rst_pdy+=( $( echo "${fnm_rst_base}" | awk -F"." '{ print $1 }' ) )
+    fnm_rst_hms+=( $( echo "${fnm_rst_base}" | awk -F"." '{ print $2 }' ) )
+  done
+
+  # Find latest restart hour
+  fnm_rst_pdy_max=$( printf '%s\n' "${fnm_rst_pdy[@]}" | sort -nu | tail -1 )
+  fnm_rst_hms_max=$( printf '%s\n' "${fnm_rst_hms[@]}" | sort -nu | tail -1 )
+ 
+  # Compute start hour of restart run for model_configure
+  cdate_org=$( $DATE_UTIL --utc --date "${PDY} ${cyc} UTC" "+%s" )
+  cdate_res=$( $DATE_UTIL --utc --date "${fnm_rst_pdy_max} ${fnm_rst_hms_max:0:2} UTC" "+%s" )
+  FHROT=$(( ( cdate_res - cdate_org )/(60*60) ))
+
+  # Create soft-link of restart files in INPUT directory
+  cd_vrfy ${DATA}/INPUT
+
+  fileids=( "coupler.res" "fv_core.res.nc" "fv_core.res.tile1.nc" "fv_srf_wnd.res.tile1.nc" "fv_tracer.res.tile1.nc" "phy_data.nc" "sfc_data.nc" )
```

Comment:
`fileids` is a bit tough to read. Please consider separating the words with an underscore`file_ids`.

Reply:
Updated.